packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -239,23 +257,22 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 				this.detachedRevision,
 			);
 			this.editManager.advanceMinimumSequenceNumber(newRevision);
+		} else {
+			const message = this.messageCodec.encode(
+				{
+					commit,
+					sessionId: this.editManager.localSessionId,
+				},
+				{
+					schema: {
+						// Clone the schema to ensure that during resubmit the schema has not been mutated by later changes
+						schema: schemaAndPolicy.schema.clone(),
```

Comment:
Is this actually necessary? `encode` seems like the kind of function that would deep-copy everything, by contract. If that's not actually guaranteed, this is fine - clone is cheap.

Also, the comment is a tiny bit confusing. Reading the comment you'd expect the clone to be happening on line 274 (it just happens to not matter since we only pass the schemaPolicy to exactly two places, so a clone in either will do.)

Reply:
done